packages/devextreme-angular/tests/src/http/ajax.spec.ts

```diff
@@ -0,0 +1,301 @@
+
+interceptors.interceptorFn = () => {};
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000;
+
+@Injectable()
+export class TestInterceptor implements HttpInterceptor {
+  intercept(req: HttpRequest<any>, next: HttpHandler) {
+    interceptors.interceptorFn();
+    return next.handle(req);
+  }
+}
+
+const createBlobFile = function (name: string, size: number) {
+  return {
+    name,
+    size,
+    type: 'image/png',
+    lastModifiedDate: new Date().getTime(),
+    blob: new Blob(['0'.repeat(size)], { type: 'application/octet-binary' }),
+    slice(startPos: number, endPos: number) {
+      return this.blob.slice(startPos, endPos);
+    },
+  };
+};
+
+@Component({
+  selector: 'test-container-component',
+  template: `<dx-file-uploader
+      uploadUrl="https://js.devexpress.com/Demos/NetCore/FileUploader/Upload"
+      [chunkSize]="200000"
+      [multiple]="true"
+      [(value)]="value"
+      accept="*"
+      uploadMode="instantly"></dx-file-uploader>
+`,
+})
+class TestFileUploaderComponent {
+  @ViewChild(DxFileUploaderComponent) fileUploader: DxFileUploaderComponent;
+
+  value = [createBlobFile('image1.png', 50100)];
+}
+
+describe('Ajax request using DxHttpModule', () => {
+  let httpTestingControllerMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [TestFileUploaderComponent],
+      imports: [HttpClientTestingModule, DxHttpModule, DxFileUploaderModule, BrowserTransferStateModule],
+      providers: [{ provide: HTTP_INTERCEPTORS, useClass: TestInterceptor, multi: true }],
+    });
+
+    httpTestingControllerMock = TestBed.inject(HttpTestingController);
+  });
+
+  afterEach(() => {
+    httpTestingControllerMock?.verify();
+  });
+
+  it('request should be correctly timeouted', (done) => {
+    const url = 'http://somefakedomain1221.com/json-url';
+
+    ajax.sendRequest({
+      url,
+      timeout: 1,
+    }).fail((error) => {
+      expect(error?.statusText).toEqual('timeout');
+      done();
+    });
+
+    httpTestingControllerMock.expectOne(url);
+  });
+
+  it('should be aborted with correct status', (done) => {
+    const url = '/heavy-url';
+    const failCallback = createSpy();
+
+    const request = ajax.sendRequest({
+      url,
+      method: 'GET',
+    });
+
+    request.fail((error) => {
+      failCallback(error);
+      expect(error?.statusText).toEqual('aborted');
+      done();
+    });
+
+    request.abort();
+
+    expect(failCallback).toHaveBeenCalledTimes(1);
+    httpTestingControllerMock.expectOne(url);
+  });
+
+  it('dataSource load() should be intercepted', (done) => {
+    const interceptorFnSpy = spyOn(interceptors, 'interceptorFn');
+    const url = 'https://js.devexpress.com/Demos/WidgetsGallery/odata/HierarchicalItems';
+    const dataSource = new DataSource({
+      store: new ODataStore({
+        version: 2,
+        url,
+      }),
+    });
+
+    // eslint-disable-next-line no-void
+    void dataSource.load().then(() => {
```

Comment:
Why do we need void here? On the first glace it seems that it could be omitted

Reply:
I get error at this place from eslint